### PR TITLE
Feature/PROD-1891 | Added validation details to msdp endpoints

### PR DESCRIPTION
**Pull request by sunny-trip-ninja**
## Description
- Updated the MSDP documentation with new validation details including
  - /msdp/search/
  - /msdp/search/removehotel/
  - /msdp/detail/hotel
  - /pre-booking
  - /ota-booking
  - /ticketing
  - /cancel

## How to Test
1. Checkout this branch
2. Run  `npx @redocly/openapi-cli preview-docs tn_api_msdp_v1.yml`
3. Go to http://127.0.0.1:8080
4. You should see the validation details added for the mentioned endpoints above in `Description` under `Response Samples` > `400`
  - e.g. image

### CODE REVIEW

**Comment by mschembri-TN** [comment]
Task linked: [PROD-1891 Add validation details to MSDP docs for search](https://app.clickup.com/t/14197839/PROD-1891)

File: tn_api_msdp_v1.yml [hunk 1]
```diff
@@ -2731,3 +2723,229 @@ components:
...
+                        type: string
+                        description: Error code
+                        example: "218"
+                    message:
+                        type: string
+                        description: Contains the error message
+                        example: "MSDP requests need a minimum of one flight and one hotel in the request."
+            INVALID_MSDP_ID:
+                type: object
+                title: "Un-matching flights and hotels ID"
```
Comment (vickchungg) [inline tn_api_msdp_v1.yml]: Would "Mis-matching" be better?

Comment (sunny-trip-ninja) [reply tn_api_msdp_v1.yml]: ooh yeah totally
Addressed!

**Review by vickchungg — APPROVED** [review]

**Comment by vickchungg** [comment]
LocalQA approved!

**Review by rob-dumont — APPROVED** [review]

**Pull request merged**